backend/api/collect.py

```
@@ -0,0 +1,432 @@
+            run_collection_job,
+            collection_job.id,
+            job_request.dict()
+        )
+        
+        logger.info(f"Created collection job {job_id} with {len(job_request.subreddits)} subreddits")
+        
+        return CollectionJobResponse.from_orm(collection_job)
+        
+    except Exception as e:
+        logger.error(f"Failed to create collection job: {e}")
+        raise HTTPException(status_code=500, detail=f"Failed to create collection job: {str(e)}")
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)